linter/lints/crl/lints.go

```diff
@@ -386,120 +391,161 @@ func hasMozReasonCodes(crl *crl_x509.RevocationList) *lint.LintResult {
 // MUST include seconds (i.e., times are YYYYMMDDHHMMSSZ), even where the number
 // of seconds is zero.  GeneralizedTime values MUST NOT include fractional
 // seconds. See: https://www.rfc-editor.org/rfc/rfc5280.html#section-4.1.2.5.2
+//
+// Conforming applications MUST encode thisUpdate, nextUpdate, and cerficate
+// validity timestamps prior to 2050 as UTCTime and GeneralizedTime there-after.
+// See:
+//   - https://www.rfc-editor.org/rfc/rfc5280.html#section-5.1.2.4
+//   - https://www.rfc-editor.org/rfc/rfc5280.html#section-5.1.2.5
+//   - https://www.rfc-editor.org/rfc/rfc5280.html#section-5.1.2.6
 func hasValidTimestamps(crl *crl_x509.RevocationList) *lint.LintResult {
 	input := cryptobyte.String(crl.RawTBSRevocationList)
-	parseFail := lint.LintResult{
+	lintFail := lint.LintResult{
 		Status:  lint.Error,
 		Details: "Failed to re-parse tbsCertList during linting",
 	}
+
 	// Read tbsCertList.
 	var tbs cryptobyte.String
-	input.ReadASN1(&tbs, cryptobyte_asn1.SEQUENCE)
+	if !input.ReadASN1(&tbs, cryptobyte_asn1.SEQUENCE) {
+		return &lintFail
+	}
+
 	// Skip (optional) version.
 	if !tbs.SkipOptionalASN1(cryptobyte_asn1.INTEGER) {
-		return &parseFail
+		return &lintFail
 	}
+
 	// Skip signature.
 	if !tbs.SkipASN1(cryptobyte_asn1.SEQUENCE) {
-		return &parseFail
+		return &lintFail
 	}
+
 	// Skip issuer.
 	if !tbs.SkipASN1(cryptobyte_asn1.SEQUENCE) {
-		return &parseFail
+		return &lintFail
 	}
+
 	// Read thisUpdate.
 	var thisUpdate cryptobyte.String
 	var thisUpdateTag cryptobyte_asn1.Tag
 	if !tbs.ReadAnyASN1Element(&thisUpdate, &thisUpdateTag) {
-		return &parseFail
+		return &lint.LintResult{Status: lint.Error, Details: "Failed to read thisUpdate timestamp"}
```

Comment:
If you want to have these more-detailed messages in here that's okay, but I think it would also be fine to just have all of these return the base `lintFail` object without reporting exactly which field failed to parse.